File: app/src/main/java/org/dhis2/utils/timber/ReleaseTree.java
package org.dhis2.utils.timber;

import androidx.annotation.NonNull;

import android.util.Log;

import org.dhis2.commons.reporting.CrashReportController;
import org.jetbrains.annotations.Nullable;

import timber.log.Timber;

/**
 * QUADRAM. Created by ppajuelo on 06/06/2018.
 */

public class ReleaseTree extends Timber.Tree {

    public CrashReportController crashReportController;

    public ReleaseTree(CrashReportController crashReportController) {
        this.crashReportController = crashReportController;
    }

    @Override
    protected boolean isLoggable(@Nullable String tag, int priority) {
        // Don't log VERBOSE, DEBUG and INFO only ERROR, WARN and WTF
        return priority != Log.VERBOSE && priority != Log.DEBUG && priority != Log.INFO;
    }

    @Override
    protected void log(int priority, String tag, @NonNull final String message, final Throwable t) {
        if (isLoggable(tag, priority)) {
            Exception e = new Exception(t);
            crashReportController.trackError(e, e.getMessage());
        }
    }
}

File: app/src/main/java/org/dhis2/utils/timber/ReleaseTree.kt
package org.dhis2.utils.timber

import android.util.Log
import com.crashlytics.android.Crashlytics
import timber.log.Timber

class ReleaseTree: Timber.Tree() {
    override fun log(priority: Int, tag: String?, message: String, t: Throwable?) {
        if (isLoggable(tag, priority))
            Crashlytics.log(priority, tag, message)
    }

    override fun isLoggable(tag: String?, priority: Int): Boolean {
        return priority != Log.VERBOSE && priority != Log.DEBUG && priority != Log.INFO
    }
}